src/commonMain/kotlin/exchange.dydx.abacus/calculator/AccountCalculator.kt

```diff
@@ -19,17 +18,229 @@ class AccountCalculator(val parser: ParserProtocol) {
     ): Map<String, Any>? {
         return if (account != null) {
             val subaccounts = parser.asMap(account["subaccounts"]) ?: return account
-            val modified = account.mutable()
+            var modified = account.mutable()
             for ((subaccountNumber, subaccount) in subaccounts) {
-                val parentSubaccountNumber = subaccountNumber.toInt() % kChildSubaccountMod
+                val parentSubaccountNumber = subaccountNumber.toInt() % NUM_PARENT_SUBACCOUNTS
                 if (parentSubaccountNumber in subaccountNumbers) {
                     val key = "subaccounts.$subaccountNumber"
-                    modified.safeSet(key, subaccountCalculator.calculate(parser.asNativeMap(subaccount), configs, markets, price, periods))
+                    modified.safeSet(
+                        key,
+                        subaccountCalculator.calculate(
+                            parser.asNativeMap(subaccount),
+                            configs,
+                            markets,
+                            price,
+                            periods,
+                        ),
+                    )
                 }
             }
+            modified = groupSubaccounts(modified, markets)
             modified
         } else {
             null
         }
     }
+
+    private fun groupSubaccounts(
+        existing: Map<String, Any>,
+        markets: Map<String, Any>?
+    ): MutableMap<String, Any> {
+        val modified = existing.mutable()
+        val subaccounts = parser.asNativeMap(parser.value(existing, "subaccounts"))
+        val subaccountNumbers = subaccounts?.keys?.mapNotNull { parser.asInt(it) }?.sorted()
+        if (subaccountNumbers != null) {
+            val groupedSubaccounts = mutableMapOf<String, Any>()
+            for (subaccountNumber in subaccountNumbers) {
+                val subaccount =
+                    parser.asNativeMap(parser.value(subaccounts, "$subaccountNumber")) ?: break
+                if (subaccountNumber < NUM_PARENT_SUBACCOUNTS) {
+                    // this is a parent subaccount
+                    groupedSubaccounts["$subaccountNumber"] = subaccount
+                } else {
+                    val parentSubaccountNumber = subaccountNumber % NUM_PARENT_SUBACCOUNTS
+                    val parentSubaccount = parser.asNativeMap(
+                        parser.value(
+                            groupedSubaccounts,
+                            "$parentSubaccountNumber",
+                        ),
+                    ) ?: parser.asNativeMap(parser.value(subaccounts, "$parentSubaccountNumber"))
+                        ?: mapOf("subaccountNumber" to parentSubaccountNumber)
+
+                    val childOpenPositions =
```

Comment:
`childOpenPositions` should only have a size of 1 unless they API traded. If they are using a `childSubaccount` as a `CROSS` margined acc, should we exclude from the merged positions in the Parent?

Reply:
> The subaccount itself doesn't indicate whether the margin is isolated or cross. I don't think our UI as designed will cover all cases.

My suggestion was more for if we should ignore adding a `childSubaccount` to the `groupedAccounts` if we see that it is being used as `CROSS` margined acc. This may be difficult because we would have to ignore it for the other subaccount endpoints as well.